src/sql/QueryEditor/QueryCodeEditor.test.tsx

```diff
@@ -0,0 +1,25 @@
+import '@testing-library/jest-dom/extend-expect';
+import React from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+import { mockQuery } from './__mocks__/query';
+import { QueryCodeEditor } from './QueryCodeEditor';
+
+const props = {
+  query: mockQuery,
+  language: '',
+  onChange: jest.fn(),
+  onRunQuery: jest.fn(),
+  getSuggestions: jest.fn(),
+  getTemplateSrv: jest.fn(),
+};
+
+describe('QueryCodeEditor', () => {
+  // The editor never finishes loading in the testing environment
+  // so we cannot test a lot here
```

Comment:
yeah this is unfortunate. I've seen us find ways to mock out the editor if there's other tests you want to do although I'm not sure it's super worth it.

Reply:
probably not worth it for this case but I would love to see if that has been hacked somehow. Do you have a link?

Reply:
here's an option but again I'm not sure it's worth it! https://github.com/grafana/grafana/blob/8722c19cc396e49ab2da8142594d9bdbb8bd10af/public/app/plugins/datasource/grafana-azure-monitor-datasource/components/QueryEditor/QueryEditor.test.tsx#L14